src/SFA.DAS.IdentifyDataLocks.Web/Pages/_PauseDateDataLockRowModel.cshtml.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using SFA.DAS.IdentifyDataLocks.Domain;
+
+namespace SFA.DAS.IdentifyDataLocks.Web.Pages
+{
+    public class PauseDateDataLockRowModel : DataLockRowModel
```

Comment:
Looking at this, it seems to me that there are more differences between `PauseDateDataLockRowModel` and `DataLockRowModel` than there are similarities.  I would say this looks to be a violation of the Liskov Substitution Principle.  Trying to use `ApprenticeValue` or `IlrValue` from a `PauseDateDataLockRowModel` instance is going to result in a `NullReferenceException`.

I would consider simply keeping them separate - without the inheritance  - and duplicating the `Header` / `ActiveDataLock` fields.  The additional constructor in `ActiveDataLocks` could just be the only constructor in a stand-alone `PauseDateDataLockRowModel`; the `period` field isn't used by `PauseDateDataLockRowModel` (outside of the constructor) anyway so that isn't needed.

This makes sense to me as a data mismatch row with one value from each of "apprenticeship" and "ilr" is quite different to a row showing two values both from "apprenticeship".

Reply:
I am glad you said that, cause I was not happy with my code anyway. It completely makes sense to separate them.